src/movies.js

```diff
@@ -1,15 +1,132 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+// Iteration 1: Ordering by year - Order by year, ascending (in growing order) 
+
+
+function orderByYear (movies) {
+
+
+    const newMovies = [...movies];
+    
+    newMovies.sort(function(movieA, movieB){
+        if(movieA.year < movieB.year){
+            return -1;
+        }
+        if(movieA.year > movieB.year){
+            return 1;
+        }
+        if(movieA.year === movieB.year){
+            if(movieA.title < movieB.title){
+                return -1
+            }
+            if(movieA.title > movieB.title){
+                return 1
+            }
+            if(movieA.title === movieB.title){
+                return 0
+            }
+            return 0;
+        }
+    });
+    return newMovies
+    }
+
+
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
 
+function howManyMovies(movies) {
+	const dramaMovies = movies.filter(function(movie){
+		
+			if (movie.genre.includes("Drama") && movie.director === "Steven Spielberg"){
+			return true 	
+			} 
+	});
+	return dramaMovies.length
+}
+
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
 
+function orderAlphabetically(movies) {
```

Comment:
This function runs good and it is easy to understand. However, to improve it performance wise you could do the map before the sort. That way you will be sorting only the list of titles and not all of the movie objects.